updated conflict resolution documents for sgw (#922)
couchbaselabs/docs-sync-gateway: modules/ROOT/pages/database-management.adoc

=== Algorithm
Although fundamentally the same, the pruning algorithm works slightly differently between Sync Gateway and Couchbase Lite.

On Sync Gateway, the pruning algorithm applies to the shortest, non-tombstoned branch in the revision tree.

> The algorithm allows the branch to retain a configurable number of revisions (revs_limit) and removes all older revisions.

Reviewer: ```suggestion
On Sync Gateway, the pruning algorithm retains a configurable number of revisions (revs_limit) and removes all older revisions.
```

Note: this retains revisions, and not revision ids.

=== Controls
You can vary the number of retained revisions using the Configuration File's {configuration-schema-database--xref--revs-limit} setting.

So, for example, with a `revs_limit` of 1,000 the algorithm keeps the last 1,000 revisions in the shortest non-tombstoned branch and remove any others from that branch.

[NOTE]
---
Sync Gateway 4.x uses automatic conflict resolution with Last Write Wins strategy, allowing `revs_limit` values as low as 1.
In earlier versions, setting `revs_limit` below 100 when `allow_conflicts = true` can lead to data loss during conflict resolution due to insufficient revision history.
---

[#lbl-rtcons]
=== Constraints
The default and minimum values of `revs_limit` are dependent on whether {configuration-schema-database--xref--allow-conflicts} is True or False.

The presence of multiple unresolved conflicts in a revision tree can impair the pruning process.

> This may prevent pruning of obsolete revisions or cause premature pruning of revisions.

Reviewer: Since there are no longer obsolete revisions or allow_conflicts=true, we can remove this.

|Automatic

> |Tombstones are not automatically purged from the bucket.

Reviewer: enable_shared_bucket_access is now always on, so you could remove this, or say that 

Sync Gateway automatically prunes tombstones

Author: I've removed the section and added the sentence that SGW prunes tombstones automatically.
